### PR TITLE
WIP: Changes for new yosys

### PR DESCRIPTION
**Pull request by acomodi**
This PR requires the new [yosys master+wip ](https://github.com/SymbiFlow/yosys/pull/81) to go green.

It adds a new techmap for the carrychain and adjusts some tests, as well as the synth.tcl script which needs some parameters to be added/changed.

### CODE REVIEW

File: xc/xc7/tests/simple_ff/CMakeLists.txt [hunk 1]
```diff
@@ -51,4 +51,6 @@ add_custom_target(
   DEPENDS ${OUT_FASM}
   )
 
-add_dependencies(all_xc7_tests simple_ff_full_loc_check)
+# XXX: need to fix attribute trimming when running with abc9
```
Comment (litghost) [inline xc/xc7/tests/simple_ff/CMakeLists.txt:54]: This isn't required anymore is it?  There was a fix upstream?

File: xc/xc7/tests/srl/srl16_amc31/basys3_dffmux_top.v [hunk 2]
```diff
@@ -12,7 +12,7 @@ input  wire [15:0]  sw,
 output wire [15:0]  led
 );
 
-parameter SRL_COUNT = 4;
+parameter SRL_COUNT = 8;
```
Comment (litghost) [inline xc/xc7/tests/srl/srl16_amc31/basys3_dffmux_top.v:15]: Why is this change required?

File: xc/xc7/tests/srl/srl32_chain/basys3_top.v [hunk 3]
```diff
@@ -90,18 +90,15 @@ always @(posedge clk)
...
 // LEDs
 genvar j;
 generate for(j=0; j<8; j=j+1) begin
   if (j < 4) begin
     assign led[j  ] = (sw[1]) ? error_lat[j] : error[j];
     assign led[j+8] = srl_q[j];
   end else begin
-    assign led[j  ] = net_0;
-    assign led[j+8] = net_0;
+    assign led[j  ] = |sw;
```
Comment (litghost) [inline xc/xc7/tests/srl/srl32_chain/basys3_top.v:100]: Why is this change required?  Are these SRL changes from debugging?

File: xc/xc7/yosys/synth.tcl [hunk 4]
```diff
@@ -44,6 +44,9 @@ opt_clean
 setundef -zero -params
 stat
 
+# TODO: remove this as soon as new VTR master+wip is pushed: https://github.com/SymbiFlow/vtr-verilog-to-routing/pull/525
```
Comment (litghost) [inline xc/xc7/yosys/synth.tcl:47]: Should have an issue to track this one.

File: xc/xc7/tests/simple_ff/CMakeLists.txt (hunk 1 above)
Comment (acomodi) [reply xc/xc7/tests/simple_ff/CMakeLists.txt:54]: I need to double check this, but this should be a different issue w.r.t. the one that did not allow the BUFGCTRL or PLL to be LOCed. This issue seems to be related on the new code that handles FF for ABC in yosys.

File: xc/xc7/tests/srl/srl16_amc31/basys3_dffmux_top.v (hunk 2 above)
Comment (acomodi) [reply xc/xc7/tests/srl/srl16_amc31/basys3_dffmux_top.v:15]: Apparently, this and the other changes to SRLs were required as VPR is not able to route GND to the top level outputs.

Comment (litghost) [reply xc/xc7/tests/srl/srl16_amc31/basys3_dffmux_top.v:15]: And masking that bug is a good solution because?

File: xc/xc7/tests/simple_ff/CMakeLists.txt (hunk 1 above)
Comment (litghost) [reply xc/xc7/tests/simple_ff/CMakeLists.txt:54]: So we still need to fix yosys before we integrate it?  It still has outstanding issues that prevent it from being usable?

Comment (acomodi) [reply xc/xc7/tests/simple_ff/CMakeLists.txt:54]: It is currently usable for the regression tests within symbiflow-arch-defs, apart from this last one. I am in the process of verifying once again that the issue is still present.

File: xc/xc7/tests/srl/srl16_amc31/basys3_dffmux_top.v (hunk 2 above)
Comment (acomodi) [reply xc/xc7/tests/srl/srl16_amc31/basys3_dffmux_top.v:15]: I'll file an issue about this, and debug it. One reason I have fixed the design was to get the all_xc7 tests to completion to check whether there were other blocking issues while using the new yosys.

File: xc/xc7/tests/simple_ff/CMakeLists.txt (hunk 1 above)
Comment (acomodi) [reply xc/xc7/tests/simple_ff/CMakeLists.txt:54]: This issue is actually solved, the LOC constraint is present after synthesis. I will re-enable the test.

Comment (litghost) [reply xc/xc7/tests/simple_ff/CMakeLists.txt:54]: Awesome!

File: xc/xc7/tests/srl/srl16_amc31/basys3_dffmux_top.v (hunk 2 above)
Comment (litghost) [reply xc/xc7/tests/srl/srl16_amc31/basys3_dffmux_top.v:15]: Both ninja and make have a "-k" to avoid stopping at failures for just this purpose.  We use this feature for the vendor tests CI for exactly this reason.

**Comment by litghost** [comment]
Superseded #1577
